src/components/fx/hover.js

```diff
@@ -906,11 +912,96 @@ function createHoverText(hoverData, opts, gd) {
 
         // remove the "close but not quite" points
         // because of error bars, only take up to a space
-        hoverData = hoverData.filter(function(d) {
+        hoverData = filterClosePoints(hoverData);
+    });
+
+    function filterClosePoints(hoverData) {
+        return hoverData.filter(function(d) {
             return (d.zLabelVal !== undefined) ||
                 (d[commonAttr] || '').split(' ')[0] === t00;
         });
-    });
+    }
+
+    // Show a single hover label
+    if(['x unified', 'y unified'].indexOf(hovermode) !== -1) {
+        // Delete leftover hover labels from other hovermodes
+        container.selectAll('g.hovertext').remove();
+
+        // similarly to compare mode, we remove the "close but not quite together" points
+        hoverData = filterClosePoints(hoverData);
+
+        // mock legend
+        var mockLayoutIn = {
+            showlegend: true,
+            legend: {
+                title: {text: t0},
+                bgcolor: fullLayout.paper_bgcolor,
+                borderwidth: 1,
+                tracegroupgap: 7,
+                orientation: 'v'
+            }
+        };
+        var mockLayoutOut = {};
+        legend.supplyLayoutDefaults(mockLayoutIn, mockLayoutOut, gd._fullData, fullLayout);
```

Comment:
Could you mock `paper_bgcolor` into `mockLayoutOut` instead of passing `fullLayout` to `legend.supplyLayoutDefauls` ?

Reply:
As explained in https://github.com/plotly/plotly.js/pull/4620#discussion_r390488261, I can simply set those in `mockLayoutIn`. Changes made in 9ea8bfdab29495a2eaf96884a36dff71a38aff67

Reply:
Resolving. Done in https://github.com/plotly/plotly.js/pull/4620/commits/9ea8bfdab29495a2eaf96884a36dff71a38aff67.